lib/scryptenc/scryptenc.c

```diff
@@ -248,28 +248,32 @@ scryptenc_setup(uint8_t header[96], uint8_t dk[64],
 		/* Check logN, r, p. */
 		if ((rc = checkparams(P->maxmem, P->maxmemfrac, P->maxtime,
 		    P->logN, P->r, P->p, verbose, force)) != 0) {
+			switch(rc) {
 			/* Warn about resource limit, but suppress the error. */
-			if (rc == SCRYPT_EBIGSLOW) {
+			case SCRYPT_EBIGSLOW:
 				warn0("Warning: Explicit parameters"
 				    " might exceed memory limit");
 				warn0("Warning: Explicit parameters"
 				    " might exceed time limit");
 				rc = 0;
-			}
-			if (rc == SCRYPT_ETOOBIG) {
+				break;
+			case SCRYPT_ETOOBIG:
 				warn0("Warning: Explicit parameters"
 				    " might exceed memory limit");
 				rc = 0;
-			}
-			if (rc == SCRYPT_ETOOSLOW) {
+				break;
+			case SCRYPT_ETOOSLOW:
 				warn0("Warning: Explicit parameters"
 				    " might exceed time limit");
 				rc = 0;
-			}
-
+				break;
 			/* Provide a more meaningful error message. */
```

Comment:
Was the intention to include this in the `switch` (which I did), or leave it separate?

I think that this is sufficiently different -- changing the error message, but not suppressing the error -- that I think it's more clear as its own `if` statement.  On the other hand, it looks really weird to have `switch(rc)` immediately followed by `if (rc)`.